tests/service/test_service.yml

```diff
@@ -0,0 +1,544 @@
+    ipaservice:
+      ipaadmin_password: SomeADMINpassword
+      name: "HTTP/{{ svc_fqdn }}"
+      allow_retrieve_keytab_user:
+      - user01
+      - user02
+      allow_retrieve_keytab_group:
+      - group01
+      - group02
+      allow_retrieve_keytab_host:
+      - "{{ host1_fqdn }}"
+      - "{{ host2_fqdn }}"
+      allow_retrieve_keytab_hostgroup:
+      - hostgroup01
+      - hostgroup02
+      action: member
+      state: absent
+    register: result
+    failed_when: not result.changed
+
+  - name: Service "HTTP/{{ svc_fqdn }}" members allow_retrieve_keytab absent for users, groups, hosts and hostgroups, again.
+    ipaservice:
+      ipaadmin_password: SomeADMINpassword
+      name: "HTTP/{{ svc_fqdn }}"
+      allow_retrieve_keytab_user:
+      - user01
+      - user02
+      allow_retrieve_keytab_group:
+      - group01
+      - group02
+      allow_retrieve_keytab_host:
+      - "{{ host1_fqdn }}"
+      - "{{ host2_fqdn }}"
+      allow_retrieve_keytab_hostgroup:
+      - hostgroup01
+      - hostgroup02
+      action: member
+      state: absent
+    register: result
+    failed_when: result.changed
+
+  #
+  - name: Ensure service is absent
+    ipaservice:
+      ipaadmin_password: SomeADMINpassword
+      name: "HTTP/{{ svc_fqdn }}"
+      state: absent
+    register: result
+    failed_when: not result.changed
+
+  - name: Ensure service is absent, again
+    ipaservice:
+      ipaadmin_password: SomeADMINpassword
+      name: "HTTP/{{ svc_fqdn }}"
+      state: absent
+    register: result
+    failed_when: result.changed
+
+  # cleanup
+
+  - name: Ensure services are absent.
+    ipaservice:
+      ipaadmin_password: SomeADMINpassword
+      name:
+      - "HTTP/{{ svc_fqdn }}"
+      - HTTP/www.ansible.com
+      - HTTP/svc.ihavenodns.info
+      - HTTP/no.idontexist.local
+      state: absent
+
+  - name: Ensure host is absent
+    ipahost:
+      ipaadmin_password: SomeADMINpassword
+      name:
+      - "{{ svc_fqdn }}"
+      - "{{ host1_fqdn }}"
+      - "{{ host2_fqdn }}"
+      - www.ansible.com
+      - svc.ihavenodns.info
+      state: absent
+
+  - name: Ensure testing users are absent.
+    ipauser:
+      ipaadmin_password: SomeADMINpassword
+      name:
+      - user01
+      - user02
+      state: absent
+
+  - name: Ensure testing groups are absent.
+    ipagroup:
+      ipaadmin_password: SomeADMINpassword
+      name:
+      - group01
+      - group02
+      state: absent
+
+  - name: Ensure testing hostgroup hostgroup01 is absent.
+    ipagroup:
+      ipaadmin_password: SomeADMINpassword
+      name:
+      - hostgroup01
+      state: absent
+
+  - name: Ensure testing hostgroup hostgroup02 is absent.
+    ipagroup:
+      ipaadmin_password: SomeADMINpassword
+      name:
+      - hostgroup02
+      state: absent
+
+  # If ansible-freeipa dnsrecord module is availble, it can be used.
+  - name: DNS record is not present.
+    command: ipa dnsrecord-del --del-all "{{ ipaserver_domain }}" svc
```

Comment:
This is not needed if you are using the host module with `update_dns: yes`. See the host module description.

Reply:
Same as the previous comment.